geowave-accumulo/src/main/java/mil/nga/giat/geowave/accumulo/util/EntryIteratorWrapper.java

```diff
@@ -94,6 +94,9 @@ public boolean hasNext() {
 	public T next()
 			throws NoSuchElementException {
 		final T previousNext = nextValue;
+		if (nextValue == null) {
```

Comment:
I think this logic is correct;  intent here (and in the other iterators changed in the same pattern) was to actually throw NoSuchElementException() when .next() was called on the end element.  (It looks like we would just return null in that case instead - but normally never got there because it was guarded with hasNext())